extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -260,18 +267,39 @@ func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRes
 }
 
 // RestoreRKE2K3SSnapshot is a helper function to restore a snapshot on an RKE2 or k3s cluster. Returns error if any.
-func RestoreRKE2K3SSnapshot(client *rancher.Client, clusterName string, snapshotRestore *rkev1.ETCDSnapshotRestore, initialControlPlaneValue, initialWorkerValue string) error {
+func RestoreRKE2K3SSnapshot(client *rancher.Client, snapshotRestore *rkev1.ETCDSnapshotRestore, clusterName string) error {
 	clusterObject, existingSteveAPIObject, err := clusters.GetProvisioningClusterByName(client, clusterName, fleetNamespace)
 	if err != nil {
 		return err
 	}
 
+	clusterObject.ObjectMeta.ResourceVersion = existingSteveAPIObject.ObjectMeta.ResourceVersion
 	clusterObject.Spec.RKEConfig.ETCDSnapshotRestore = snapshotRestore
-	clusterObject.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency = initialControlPlaneValue
-	clusterObject.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency = initialWorkerValue
 
 	logrus.Infof("Restoring snapshot: %v", snapshotRestore.Name)
-	_, err = client.Steve.SteveType(ProvisioningSteveResouceType).Update(existingSteveAPIObject, clusterObject)
+	updatedCluster, err := client.Steve.SteveType(ProvisioningSteveResouceType).Update(existingSteveAPIObject, clusterObject)
+	if err != nil {
+		return err
+	}
+
+	err = kwait.PollUntilContextTimeout(context.TODO(), 500*time.Millisecond, defaults.ThirtyMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
supernit; 500ms over 30 minutes seems like a lot of polling

Reply:
also, can't you just use the function for checking if an upgrade is ready instead of writing your own?

Reply:
I'll update the minutes. v2 prov has known issues with cluster status not accurately having the test finish. Delete init node is an immediate example I can think of. I'd prefer to have this in for reassurance.